tests-fuzz/src/context.rs

```diff
@@ -48,3 +55,183 @@ impl From<&CreateTableExpr> for TableContext {
         }
     }
 }
+
+impl TableContext {
+    /// Applies the [AlterTableExpr].
+    pub fn alter(mut self, expr: AlterTableExpr) -> Result<TableContext> {
+        match expr.alter_options {
+            AlterTableOperation::AddColumn { column, location } => {
+                ensure!(
+                    !self.columns.iter().any(|col| col.name == column.name),
+                    error::UnexpectedSnafu {
+                        violated: format!("Column {} exists", column.name),
+                    }
+                );
+                match location {
+                    Some(AddColumnLocation::First) => {
+                        let mut columns = Vec::with_capacity(self.columns.len() + 1);
+                        columns.push(column);
+                        columns.extend(self.columns);
+                        self.columns = columns;
+                    }
+                    Some(AddColumnLocation::After { column_name }) => {
+                        let index = self
+                            .columns
+                            .iter()
+                            // TODO(weny): find a better way?
```

Comment:
@WenyXu why do you think this impl is suboptimal? It literally implement the semantic and since total columns should be in a small set I think it's good to do:

```rust
fn position<P>(&mut self, mut predicate: P) -> Option<usize> where ...
{
                let n = len!(self);
                let mut i = 0;
                while let Some(x) = self.next() {
                    if predicate(x) {
                        return Some(i);
                    }
                    i += 1;
                }
                None
}
```